File: Python/project/src/Lists/SeperateDigitsInArray/SeperateDigitsInArray.py
from typing import List


def main():

    input = [[13,25,83,77], [7,1,3,9]]

    for i, row in enumerate(input):
        solution = Solution()
        print("seperateDigits: " + str(solution.separateDigits(row)) + "\n")

class Solution:
    def separateDigits(self, nums: List[int]) -> List[int]:
        output = []
        print("list: " + str(nums))

        for i, num in enumerate(nums):
            numStr = str(num)
            print("     num: " + numStr)

            for j, digit in enumerate(numStr):
                print("         digit: " + str(digit))
                output.append(int(digit))



        return output


if __name__ == "__main__":
    main()

File: Java/src/main/java/Lists/SeperateDigitsInArray/SeperateDigitsInArray.java
package Lists.SeperateDigitsInArray;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class SeperateDigitsInArray {
    public static void main(String [] args) {
        int [][] input = {{13,25,83,77}, {7,1,3,9}};
        for (int [] i : input)
            System.out.println(Arrays.toString(separateDigits(i)) + "\n");
    }

    public static int[] separateDigits(int[] nums) {
        List<Integer> list = new ArrayList<>();

        for (int num : nums) {
            String numStr = String.valueOf(num);
            String [] numStrArr = numStr.split("");

            for (int i=0; i<numStrArr.length; i++) {
                list.add(Integer.valueOf(numStrArr[i]));
            }
        }

        int [] output = new int[list.size()];
        for (int i=0; i<list.size(); i++){
            output[i] = list.get(i);
        }


        return output;
    }
}
